hamlet/celery.py

```diff
@@ -346,30 +407,54 @@ def ml_subject_assistant_export_to_microsoft(
         ) as out_f:
             json.dump(data, out_f)
             out_f.flush()
-            out_f_name = out_f.name
-
-        # Save the created file to the database
-        with open(out_f_name, 'rb') as out_f:
-            export.json.save(
-                'ml-subject-assistant-{}-export{}.json'.format(
-                    export.subject_set_id,
-                    export.id,
-                ),
-                File(out_f),
-            )
-
-        # SUCCESS
-        export.status = MLSubjectAssistantExport.COMPLETE
-        export.save()              
+            source_filepath = out_f.name
+        
+        return source_filepath
     
-    except Exception as e:
+    except Exception as err:
+        # Only cleanup the temporary file on error; otherwise it'll be cleaned up in the main function.
         try:
-            self.retry(countdown=60)
-        except MaxRetriesExceededError:
-            export.status = MLSubjectAssistantExport.FAILED
-            export.save()
-            raise e
-    
-    finally:
-        os.unlink(out_f_name)
+            if len(source_filepath) > 0:
+                os.unlink(source_filepath)
+        except OSError:
+            pass
+      
+        raise err
+  
+def ml_subject_assistant_export_to_microsoft_pt3_create_shareable_azure_blob(
+    source_filepath,
+    target_filename,
+):
+    shareable_file_url = ''
+  
+    try:
+        azure_account_name = os.environ.get('SUBJECT_ASSISTANT_AZURE_ACCOUNT_NAME')
```

Comment:
These should probably be moved into settings.py (which can just load them from the environment too, but it'll make it easier to configure the settings all in one place if we want to). See for example the [AWS settings](https://github.com/zooniverse/hamlet/blob/master/hamlet/settings.py#L178).

Reply:
👌